zerver/lib/rest.py

```diff
@@ -69,7 +69,9 @@ def rest_dispatch(request, globals_list, **kwargs):
         #
         # Security implications of this portion of the code are minimal,
         # as we should worst-case fail closed if we miscategorise a request.
-        if not request.path.startswith("/api") and request.user.is_authenticated():
+        # Existance of both '/api' and 'HTTP_AUTHORIZATION is checked to that'
+        # files uploaded through API are also accesible via web clients.
+        if not(request.path.startswith("/api") and request.META.get('HTTP_AUTHORIZATION', None)) and request.user.is_authenticated():
```

Comment:
@timabbott Had to add that extra condition. Otherwise files uploaded via api were not accessible through the web client